prac_10/wiki.py

```diff
@@ -0,0 +1,39 @@
+"""Display summaries of Wikipedia pages from user input."""
+
+import wikipedia
+
+FIRST_RESULT_INDEX = 0
+
+
+def main():
+    """Display summaries of Wikipedia pages from user input."""
+    wiki_input = input("Enter page title or search phrase:")
+    while wiki_input != "":
+        wiki_pages = get_wiki_pages_from_input(wiki_input)
+        display_information_of_wiki_page(wiki_pages)
+        wiki_input = input("Enter page title or search phrase:")
+
+
+def get_wiki_pages_from_input(wiki_input):
+    """Search Wikipedia for page title or phrase and return Wikipedia pages list."""
+    wiki_pages = wikipedia.search(wiki_input, results=10)
+    return wiki_pages
+
+
+def display_information_of_wiki_page(wiki_pages: list):
+    """
+    Display title, summary, and URL of first Wikipedia page in the given list of pages,
+    that is not a Disambiguation page.
+    """
+    wiki_page_title = wiki_pages[FIRST_RESULT_INDEX]
+    try:
+        wiki_page = wikipedia.page(wiki_page_title, auto_suggest=False)
+        print("Title: {}\n"
+              "Summary:\n{}\n"
+              "URL:{}".format(wiki_page.title, wiki_page.summary.strip(), wiki_page.url))
```

Comment:
Is this the most efficient way to do this? If we know that the summary returns a \n at the end, why make extra work of adding the \n in the print statement and strip it from the returned summary?